### PR TITLE
Refactor cache date

### PR DESCRIPTION
**Pull request by shimonewman**
## What does this PR do?

(Provide a description of what this PR does.)

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work.)

## Related PRs and Issues

(If this PR is related to any other PR or resolves any issue or related to any issue link all related PR and issues here.)

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

(Write your answer here.)

### CODE REVIEW

File: app/controllers/shared/api.php [hunk 1]
```diff
@@ -358,30 +358,33 @@
...
                     'resource' => $resource,
-                    'accessedAt' => \time(),
+                    'accessedAt' => $now,
                     'signature' => $signature,
                     ])));
-                } elseif (date('Y/m/d', \time()) > date('Y/m/d', $cacheLog->getAttribute('accessedAt'))) {
-                    $cacheLog->setAttribute('accessedAt', \time());
+                } elseif (date('Y/m/d', \time()) > date('Y/m/d', strtotime($cacheLog->getAttribute('accessedAt')))) {
+                    var_dump('update');
+                    var_dump($now);
```
Comment (christyjacob4) [inline app/controllers/shared/api.php:386]: we should remove these

File: app/controllers/shared/api.php [hunk 2]
```diff
@@ -358,30 +358,33 @@
...
                 $key = md5($request->getURI() . implode('*', $request->getParams()));
-
                 $data = json_encode([
                 'content-type' => $response->getContentType(),
                 'payload' => base64_encode($data['payload']),
                 ]) ;
 
                 $signature = md5($data);
                 $cacheLog  = $dbForProject->getDocument('cache', $key);
+                $now = date('d-m-Y H:i:s', time());
```
Comment (christyjacob4) [inline app/controllers/shared/api.php:376]: With the new `datetime` changes we should use the Datetime class to get the current time. 

Reference 
https://github.com/appwrite/appwrite/blob/0.16.x/app/controllers/general.php#L310

File: app/controllers/shared/api.php [hunk 3]
```diff
@@ -358,30 +358,33 @@
...
                     Authorization::skip(fn () => $dbForProject->createDocument('cache', new Document([
                     '$id' => $key,
                     'resource' => $resource,
-                    'accessedAt' => \time(),
+                    'accessedAt' => $now,
                     'signature' => $signature,
                     ])));
-                } elseif (date('Y/m/d', \time()) > date('Y/m/d', $cacheLog->getAttribute('accessedAt'))) {
-                    $cacheLog->setAttribute('accessedAt', \time());
+                } elseif (date('Y/m/d', \time()) > date('Y/m/d', strtotime($cacheLog->getAttribute('accessedAt')))) {
```
Comment (christyjacob4) [inline app/controllers/shared/api.php:384]: For comparisons as well. Let's use the new datetime class

File: app/controllers/shared/api.php (hunk 2 above)
Comment (shimonewman) [reply app/controllers/shared/api.php]: fixed

File: app/controllers/shared/api.php (hunk 1 above)
Comment (shimonewman) [reply app/controllers/shared/api.php]: fixed

File: app/controllers/shared/api.php (hunk 3 above)
Comment (shimonewman) [reply app/controllers/shared/api.php]: fixed, using the DateTime::now() still needs to be formatted

File: app/controllers/shared/api.php [hunk 4]
```diff
@@ -362,30 +363,33 @@
...
                 $cacheLog  = $dbForProject->getDocument('cache', $key);
+                $accessedAt = $cacheLog->getAttribute('accessedAt', '');
+                $accessedAt = (new \DateTime($accessedAt))->format('Y/m/d');
+
                 if ($cacheLog->isEmpty()) {
                     Authorization::skip(fn () => $dbForProject->createDocument('cache', new Document([
                     '$id' => $key,
                     'resource' => $resource,
-                    'accessedAt' => \time(),
+                    'accessedAt' => DateTime::now(),
```
Comment (christyjacob4) [inline app/controllers/shared/api.php:388]:
```suggestion
                    'accessedAt' => \DateTime(),
```


`DateTime::now()` returns a formatted string. We need to use the DateTime object directly as it then goes through the database filters

File: app/controllers/shared/api.php [hunk 5]
```diff
@@ -362,30 +363,33 @@
...
-
                 $data = json_encode([
                 'content-type' => $response->getContentType(),
                 'payload' => base64_encode($data['payload']),
                 ]) ;
 
                 $signature = md5($data);
                 $cacheLog  = $dbForProject->getDocument('cache', $key);
+                $accessedAt = $cacheLog->getAttribute('accessedAt', '');
+                $accessedAt = (new \DateTime($accessedAt))->format('Y/m/d');
```
Comment (christyjacob4) [inline app/controllers/shared/api.php:382]: Why aren't we using this variable in Line 388 ?

File: app/controllers/shared/api.php [hunk 6]
```diff
@@ -362,30 +363,33 @@
...
                     Authorization::skip(fn () => $dbForProject->createDocument('cache', new Document([
                     '$id' => $key,
                     'resource' => $resource,
-                    'accessedAt' => \time(),
+                    'accessedAt' => DateTime::now(),
                     'signature' => $signature,
                     ])));
-                } elseif (date('Y/m/d', \time()) > date('Y/m/d', $cacheLog->getAttribute('accessedAt'))) {
-                    $cacheLog->setAttribute('accessedAt', \time());
+                } elseif ((new \DateTime(DateTime::now()))->format('Y/m/d') > $accessedAt) {
```
Comment (christyjacob4) [inline app/controllers/shared/api.php:391]: Let's be more consistent and use something like this 
https://github.com/appwrite/appwrite/blob/0.16.x/app/controllers/general.php#L310

File: app/workers/deletes.php [hunk 7]
```diff
@@ -225,7 +225,7 @@ protected function deleteUsageStats(string $datetime1d, string $datetime30m)
             ], $dbForProject);
 
             $this->deleteByGroup('stats', [
-                Query::lessThan('time', [$datetime30m]),
+                Query::lessThan('time', $datetime30m),
```
Comment (lohanidamodar) [inline app/workers/deletes.php:228]: This doesn't make sense? being array or not it should be same right? we have the same below `Query::equal('period', ['30m']`?

However, the change should not make a difference as far as I know how database query works

Comment (shimonewman) [reply app/workers/deletes.php:228]: The DateTime validator was throwing errors.
The diff is period is a string  while time is dateTime attr

**Pull request merged**